python/tank/bootstrap/resolver.py

```diff
@@ -25,139 +25,102 @@
 
 log = LogManager.get_logger(__name__)
 
+
 class ConfigurationResolver(object):
     """
-    Base class for defining recipes for how to resolve a configuration object
-    given a particular project and configuration.
+    A class that contains the business logic for returning a configuration
+    object given a set of parameters.
     """
 
-    def __init__(self, sg_connection, bundle_cache_fallback_paths):
-        """
-        Constructor.
-
-        :param sg_connection: Shotgun API instance
-        :param bundle_cache_fallback_paths: List of additional paths where apps are cached.
-        """
-        self._sg_connection = sg_connection
-        self._bundle_cache_fallback_paths = bundle_cache_fallback_paths
-
-    def resolve_configuration(
-        self,
-        project_id,
-        pipeline_config_name,
-        engine_name,
-        base_config_descriptor,
-        get_latest_config
+    def __init__(
+            self,
+            project_id,
+            entry_point,
+            engine_name,
+            bundle_cache_fallback_paths
```

Comment:
Is there any reason to make bundle_cache_fallback_paths required?  Would a default of ```[]``` make sense?